images/capi/packer/gce/scripts/ci-gce.sh

```diff
@@ -0,0 +1,106 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+################################################################################
+# usage: ci-gce.sh
+# This program build all images for capi gce
+################################################################################
+
+set -o nounset
+set -o pipefail
+
+CAPI_ROOT=$(dirname "${BASH_SOURCE[0]}")/../../..
+cd "${CAPI_ROOT}" || exit 1
+
+# Verify the required Environment Variables are present.
+: "${GOOGLE_APPLICATION_CREDENTIALS:?Environment variable empty or not defined.}"
+
+export GCP_REGION=${GCP_REGION:-"us-east4"}
+export GCP_B64ENCODED_CREDENTIALS=$(base64 -w0 "$GOOGLE_APPLICATION_CREDENTIALS")
+
+cleanup() {
+   # removing the image created
+  (gcloud compute images list --project "$GCP_PROJECT" \
+    --no-standard-images --filter="family:capi-ubuntu-*" --format="table[no-heading](name)" \
+        | awk '{print "gcloud compute images delete --project '"$GCP_PROJECT"' --quiet " $1 "\n"}' \
+        | bash) || true
+
+  # stop boskos heartbeat
+  [[ -z ${HEART_BEAT_PID:-} ]] || kill -9 "${HEART_BEAT_PID}"
+}
+trap cleanup EXIT
+
+# If BOSKOS_HOST is set then acquire an GCP account from Boskos.
+if [ -n "${BOSKOS_HOST:-}" ]; then
+  # Check out the account from Boskos and store the produced environment
+  # variables in a temporary file.
+  account_env_var_file="$(mktemp)"
+  python hack/checkout_account.py 1>"${account_env_var_file}"
+  checkout_account_status="${?}"
+
+  # If the checkout process was a success then load the account's
+  # environment variables into this process.
+  # shellcheck disable=SC1090
+  [ "${checkout_account_status}" = "0" ] && . "${account_env_var_file}"
+
+  # Always remove the account environment variable file. It contains
+  # sensitive information.
+  rm -f "${account_env_var_file}"
+
+  if [ ! "${checkout_account_status}" = "0" ]; then
+    echo "error getting account from boskos" 1>&2
+    exit "${checkout_account_status}"
+  fi
+
+  # run the heart beat process to tell boskos that we are still
+  # using the checked out account periodically
+  python -u hack/heartbeat_account.py >> "$ARTIFACTS/logs/boskos.log" 2>&1 &
+  HEART_BEAT_PID=$(echo $!)
+fi
+
+if [[ -z "$GOOGLE_APPLICATION_CREDENTIALS" ]]; then
+  cat <<EOF
+GOOGLE_APPLICATION_CREDENTIALS is not set.
+Please set this to the path of the service account used to run this script.
+EOF
+  return 2
+else
+  gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+fi
+if [[ -z "$GCP_PROJECT" ]]; then
+  GCP_PROJECT=$(jq -r .project_id "${GOOGLE_APPLICATION_CREDENTIALS}")
+  cat <<EOF
+GCP_PROJECT is not set. Using project_id $GCP_PROJECT
+EOF
+fi
+if [[ -z "$GCP_REGION" ]]; then
+  cat <<EOF
+GCP_REGION is not set.
+Please specify which the GCP region to use.
+EOF
+  return 2
+fi
+
+export GCP_PROJECT_ID="$GCP_REGION"
```

Comment:
This bit doesn't seem right...  Should this be set to `GCP_PROJECT` instead?
```suggestion
export GCP_PROJECT_ID="$GCP_PROJECT"
```

Reply:
nice catch!
copy/paste issues lol

Reply:
done